editor/lang/token.hh

```diff
@@ -33,20 +31,78 @@ public:
     // Other
     NUMBER,
     IDENTIFIER,
+    NONE
   };
 
   Token(Type type, int line, std::string lexeme, std::string literal)
       : m_type(type), m_line(line), m_lexeme(lexeme), m_literal(literal) {}
 
+  Token() : Token(NONE, 0, "", "") {}
+
+  Token(Type type, std::string lexeme) : m_type(type), m_lexeme(lexeme) {}
+
   friend std::ostream &operator<<(std::ostream &stream, const Token &token) {
-    stream << token.m_lexeme << "," << magic_enum::enum_name(token.m_type);
+	stream << token.m_lexeme << "," << token.m_type;
 
     return stream;
   }
 
+  void set_token_type(Type type) {
```

Comment:
I think this function shouldn't do anything instead, another `static` one would take the `Type` and convert it to a string using a Hash Map.